### PR TITLE
AB2-761 AB-test for go settings button

### PR DESCRIPTION
**Pull request by giota-cliqz**

### CODE REVIEW

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java [hunk 1]
```diff
@@ -140,10 +147,23 @@ public ActivityStreamPanel(final Context context, AttributeSet attrs) {
...
 
         final View customizeTabLink = customizeNewTabView.findViewById(R.id.customize_tab_link);
+        if (abManager.getHomeSettingsStyle() == "text") {
+            customizeTabLink.setBackgroundColor(Color.argb(0, 0, 0, 0));
+        }
         customizeTabLink.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 final Intent intent = new Intent(context, GeckoPreferences.class);
+                final GeckoBundle signal = new GeckoBundle();
```
Comment (giota-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: I need to add similar signal on load. Where should I add it?

Comment (spacifici) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: Why?

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java [hunk 2]
```diff
@@ -140,10 +147,23 @@ public ActivityStreamPanel(final Context context, AttributeSet attrs) {
...
+                signal.putString("type", "onboarding");
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: How many telemetry signal will have the same type? More than 1 -> we should generalize

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java [hunk 3]
```diff
@@ -140,10 +147,23 @@ public ActivityStreamPanel(final Context context, AttributeSet attrs) {
...
+                signal.putString("action", "click");
+                signal.putString("target", "settings");
+                signal.putString("target", "home_customization");
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: You are overwriting the target

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java [hunk 4]
```diff
@@ -140,10 +147,23 @@ public ActivityStreamPanel(final Context context, AttributeSet attrs) {
...
+                signal.putString("style", "text");
+                if (abManager.getHomeSettingsStyle() == "button") {
+                    signal.putString("style", "button");
+                }
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]:
```java
signal.putString("style", ABManager.VALUE_BUTTON.equals(abManager.getHomeSettingsStyle() ? 
    ABManager.VALUE_BUTTON : ABManager.VALUE_TEXT);
```

Comment (spacifici) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: All the string constants have to be replaced by constants.

File: mozilla-release/mobile/android/base/java/com/cliqz/ABManager.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        if (abManager == null) {
+            abManager = new ABManager(context.getApplicationContext());
+        }
+    }
+
+    public static ABManager getInstance() {
+        return abManager;
+    }
+
+    public String getHomeSettingsStyle() {
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/com/cliqz/ABManager.java]: Mark this method as ```@NonNull```.

File: mozilla-release/mobile/android/base/java/com/cliqz/ABManager.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import org.mozilla.gecko.GeckoSharedPrefs;
+import org.mozilla.gecko.PrefsHelper;
+
+
+public class ABManager {
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/com/cliqz/ABManager.java]:
```java
/**
 * Copyright © Cliqz 2019
 */
```

Comment (spacifici) [reply mozilla-release/mobile/android/base/java/com/cliqz/ABManager.java]: Description of what this is about will be helpful

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java (hunk 1 above)
Comment (giota-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: In the ticket https://cliqztix.atlassian.net/browse/AB2-667 there are 2 signals I need to add, one on click and one on show. Is the constructor the right place to send the telemetry for show action?

Comment (spacifici) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: @ravjit-cliqz you implemented the auto disappearing thing, can you help here?

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java (hunk 2 above)
Comment (giota-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java]: `sendTelemetry` accepts a Geckobundle.

We will have 2 signals, so I will generalize. Shall I create a GoSettingsTelemetry class? Or maybe a HomePageTelemetry? Where should I create this file?

**Pull request merged**